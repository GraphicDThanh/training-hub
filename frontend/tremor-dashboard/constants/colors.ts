type ColorChannelsChartType = {
  [key: number]: string;
  1: string;
  2: string;
  3: string;
  4: string;
};

export const COLOR_CHANNELS_CHART: ColorChannelsChartType = {
  1: "blue-500",
  2: "red-500",
  3: "primary",
  4: "gray-500",
};
