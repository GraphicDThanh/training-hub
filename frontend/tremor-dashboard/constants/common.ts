// Types
import { OptionType } from "@/types";

export const EXCEPT_KEYS = {
  POSITIVE_INTEGER: ["e", "E", "+", "-", "."],
  POSITIVE_DOUBLE: ["e", "E", "+", "-"],
};

export enum DIRECTION {
  ASC = "asc",
  DESC = "desc",
}

// Mailto Support
export const MAILTO_SUPPORT = "support@tremor-dashboard.com";

export const REMEMBER_ME_COOKIES_KEY = "remember-me";
export const UID_KEY = "uid";
export const ROLE = "role";
export const NAME = "name";
export const AVATAR = "avatar";
export const AUTH_SESSION_COOKIES_KEY = "authjs.session-token";

export const ORDER_LIST_OPTIONS: OptionType[] = [
  { option: "Paid", value: "0" },
  { option: "Refunded", value: "2" },
  { option: "Canceled", value: "1" },
];

export const PRODUCT_LIST_OPTIONS: OptionType[] = [
  { option: "Yes", value: "true" },
  { option: "No", value: "false" },
];

export const FORMAT_DATE = {
  YEAR_MONTH_DATE_TIME: "YYYY-MM-DDTHH:mm:ss.SSS",
  MONTH_DAY_YEAR_TIME: "MMM D, YYYY h:mm A",
};

export const SEPARATOR = {
  COMMAS: ",",
  DOTS: ".",
};

export const CDN_KEY = process.env.NEXT_PUBLIC_CDN_KEY;

export const DRAG_ZONE = {
  DEFAULT: {
    TEXT: "Drag and drop your file here to begin uploading.",
    STYLE_INPUT: "border-gray-300",
    STYLE_TEXT: "text-gray-500 dark:text-gray-400",
  },
  ON_DRAG: {
    TEXT: "Release the mouse click to upload the photo",
    STYLE_INPUT: "outline outline-2 outline-[#ebeff4]",
    STYLE_TEXT: "text-gray-500 dark:text-gray-400",
  },
  ON_WRONG_FORMAT: {
    TEXT: "Invalid image file type. Please try uploading again.",
    STYLE_INPUT: "border-red-500",
    STYLE_TEXT: "text-red-500 dark:text-red-500",
  },
};

export const TRANSACTION_CLASS = {
  INCREASE: {
    buttonClasses: "text-few dark:text-few dark:hove:text-few border-few",
    textClasses: "font-semibold text-greener dark:text-few",
  },
  DECREASE: {
    buttonClasses:
      "text-attention dark:text-attention dark:hover:text-attention border-attention",
    textClasses: "font-semibold text-blood dark:text-pink",
  },
  PENDING: {
    buttonClasses:
      "text-primary dark:text-primary dark:hover:text-primary border-primary",
    textClasses:
      "font-semibold text-primary dark:text-dark-tremor-content-pending",
  },
};

export const enum VARIANT_BUTTON {
  LIGHT = "light",
  LIGHT_CARD = "lightCard",
  LIGHT_CENTER = "lightCenter",
  LIGHT_STATUS = "lightStatus",
  SECONDARY = "secondary",
  SECONDARY_SHADOW = "secondaryShadow",
  PRIMARY = "primary",
  PRIMARY_CENTER = "primaryCenter",
  SURFACE = "surface",
  DARK = "dark",
  ACTIVE = "active",
}

export const NO_IMAGE = "No Image";

export const RESULT_NOT_FOUND = "Result Not Found";

export const AMOUNT_LIMIT = 20000;
export const PRICE_LIMIT = 20000;
export const QUANTITY_LIMIT = 10000;
export const WEIGHT_LIMIT = 2000;

export const PREFIX_PHONE_NUMBER = "+1";

export const STORAGES = {
  IS_DARK_THEME: "isDarkTheme",
};

export const THEMES = {
  LIGHT: "light",
  DARK: "dark",
};

export const LOGO_TITLE = "Tremor Dashboard PRO";
