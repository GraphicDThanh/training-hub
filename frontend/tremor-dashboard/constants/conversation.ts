export const CONVERSATION_TABLES = {
  GROUPS: "groups",
  CHATS: "chats",
};

export const CONVERSATION_ORDERS = {
  CREATED_AT: "createdAt",
  UPDATED_AT: "updatedAt",
};

export const CONVERSATION_QUERYS = {
  USER_IDS: "userIds",
};

export enum CONVERSATION_STATUS {
  INITIAL = "initial",
  FETCHED = "fetched",
}

export const ADMIN_ID = parseInt(process.env.NEXT_PUBLIC_ADMIN_ID!);

export enum DOCUMENT_CHANGE_TYPE {
  ADDED = "added",
  MODIFIED = "modified",
  REMOVED = "removed",
}

export enum WHERE_FILTER_OP {
  ARRAY_CONTAINS = "array-contains",
}
