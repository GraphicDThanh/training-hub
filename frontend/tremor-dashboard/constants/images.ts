export const NOT_FOUND_IMAGE = "/images/no-image-square.webp";
export const LOGO_IMAGE = "/assets/images/sidebar-logo.webp";
export const PLACEHOLDER_IMAGE = "/images/image-regular.svg";

export const AVATAR_IMAGE = {
  XS: "/images/avatar/avatar-xs.webp",
  SM: "/images/avatar/avatar-sm.webp",
  MD: "/images/avatar/avatar-md.webp",
  LG: "/images/avatar/avatar-lg.webp",
};

export const METADATA_IMAGE = {
  HOME_PREVIEW: "/assets/images/home-preview.webp",
  MASTERCARD: "/assets/images/mastercard.webp",
};

export const PROJECT_IMAGE = "/assets/images/profile-project.webp";
