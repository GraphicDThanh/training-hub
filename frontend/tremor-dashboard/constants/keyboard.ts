export const KEYBOARD_KEYS = {
  BACKSPACE: "Backspace",
  DIGIT_1: "1",
};
