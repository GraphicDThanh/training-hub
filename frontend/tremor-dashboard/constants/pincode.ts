export const PIN_CODE_DEFAULT = "*";
export const PIN_CODE_LENGTH = 6;
export const PIN_CODE_SKIP_SET_KEY = "skipSetupPinCode";
