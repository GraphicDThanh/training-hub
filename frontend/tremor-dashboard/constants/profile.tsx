import { PlatformSetting } from "@/types";

export const SETTING_FIELDS_ACCOUNT: PlatformSetting[] = [
  { label: "Email me when someone follows me", field: "emailMentions" },
  {
    label: "Email me when someone answers on my post",
    field: "emailFollowing",
  },
  { label: "Email me when someone mentions me", field: "emailAnswerPost" },
];

export const SETTING_FIELDS_APPLICATION: PlatformSetting[] = [
  { label: "New launches and projects", field: "newLaunchesProject" },
  { label: "Monthly product updates", field: "monthlyProductUpdate" },
  { label: "Subscribe to newsletter", field: "subscribeToNewsletter" },
];

export const SETTING_DATA_ACCOUNT = {
  emailMentions: false,
  emailFollowing: false,
  emailAnswerPost: false,
};

export const SETTING_DATA_APPLICATION = {
  newLaunchesProject: false,
  monthlyProductUpdate: false,
  subscribeToNewsletter: false,
};

// Social links
export const SOCIAL_LINKS = {
  facebook: "https://www.facebook.com/",
  twitter: "https://twitter.com/",
  instagram: "https://instagram.com",
};
