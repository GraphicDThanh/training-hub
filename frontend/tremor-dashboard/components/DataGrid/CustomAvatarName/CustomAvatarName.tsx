"use client";

import Modal from "@/components/Modal/Modal";
// Constants
import { NOT_FOUND_IMAGE } from "@/constants";

// Components
import { Text } from "@tremor/react";
import Image from "next/image";
import { useState } from "react";

interface CustomAvatarNameProps {
  avatar?: string;
  text: string;
  enableMagnify?: boolean;
}

const CustomAvatarName = ({
  avatar,
  text,
  enableMagnify,
}: CustomAvatarNameProps) => {
  const [avatarURL, setAvatarURL] = useState(avatar ?? NOT_FOUND_IMAGE);
  const [isMagnify, setIsMagnify] = useState(false);

  const handleInvalidImage = () => {
    setAvatarURL(NOT_FOUND_IMAGE);
  };

  const handleMagnify = () => {
    setIsMagnify(prev => !prev);
  };

  const Container = enableMagnify ? "button" : "div";

  return (
    <div className="flex justify-start items-center gap-2">
      <Container className="relative h-6 w-6" onClick={handleMagnify}>
        <Image
          src={avatarURL || NOT_FOUND_IMAGE}
          alt={`${text} avatar`}
          fill
          sizes="24px"
          className="object-cover rounded-full"
          onError={handleInvalidImage}
        />
      </Container>

      {enableMagnify && isMagnify && (
        <Modal
          additionalClasses="w-[100%] gap-5 pb-0 dark:bg-dark-tremor-primary"
          showCloseIconBtn
          showBtnCloseLabel={false}>
          <Image
            src={avatarURL}
            alt="avatar"
            width={700}
            height={400}
            className="object-fill p-2"
          />
        </Modal>
      )}

      <Text className="text-xs text-tertiary dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] capitalize order-customer w-[200px] truncate">
        {text}
      </Text>
    </div>
  );
};
export default CustomAvatarName;
