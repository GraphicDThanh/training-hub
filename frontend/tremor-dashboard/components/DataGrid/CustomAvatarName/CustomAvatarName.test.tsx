import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

// Constants
import { AVATAR_IMAGE } from "@/constants";

// Components
import CustomAvatarName from "./CustomAvatarName";

describe("CustomAvatarName", () => {
  const props = {
    text: "John Doe",
    href: "",
  };

  const renderCustomAvatarName = () =>
    render(<CustomAvatarName avatar={AVATAR_IMAGE.XS} {...props} />);

  it("should match snapshot", () => {
    const { container } = renderCustomAvatarName();

    expect(container).toMatchSnapshot();
  });

  it("renders the correct src", () => {
    const { getByRole } = render(<CustomAvatarName {...props} />);

    const image = getByRole("img");
    const urlImage = image.getAttribute("src");
    expect(urlImage).toBeTruthy();
  });
});
