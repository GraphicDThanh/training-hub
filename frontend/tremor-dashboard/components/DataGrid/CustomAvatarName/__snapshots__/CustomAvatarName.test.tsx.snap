// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomAvatarName should match snapshot 1`] = `
<div>
  <div
    class="flex justify-start items-center gap-2"
  >
    <div
      class="relative h-6 w-6"
    >
      <img
        alt="John Doe avatar"
        class="object-cover rounded-full"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="24px"
        src="http://localhost/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75"
        srcset="/_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=16&q=75 16w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=32&q=75 32w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=48&q=75 48w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=64&q=75 64w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=96&q=75 96w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=128&q=75 128w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Favatar%2Favatar-xs.webp&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </div>
    <p
      class="text-tremor-default text-xs text-tertiary dark:text-lighter font-semibold leading-[15px] tracking-[0.4px] capitalize order-customer w-[200px] truncate"
    >
      John Doe
    </p>
  </div>
</div>
`;
